feature-configs/5g-ref-nw/README.md

```diff
@@ -0,0 +1,140 @@
+- Segregation between the common template and the specific configuration of the particular cluster instance.
+    - This is derived from the assumption that many clusters will use the same hardware configuration, and only differ by a small amount of network settings.
+- Reuse of components and duplication avoidance
+    - We want to avoid the duplication of the same code fragment in different places around the product. However, this sometimes comes at a price of making the code structure less intuitive for understanding. This is especially true when using configuration overlays.
+- Usability /scalability considerations
+    - Assuming that production deployments may have thousands of nodes organized in hundreds of clusters, we want the structure to facilitate the following use cases:
+        - Create a new cluster based on an existing template
+        - Compare two clusters
+        - Add a new role (in addition to CU and DU)
+        - Add a new hardware profile
+
+## Folder hierarchy
+
+<img src="images/hierarchy.png">
+
+### __operators__
+The [operators](./operators) folder contains links to the operator deployments in the project' [`deploy`](../deploy) folder. 
+
+TODO: Currently only performance addon, ptp and sr-iov operators are defined here. Add local storage operator
+
+### __profile-base__
+The [profile-base](./profile-base) folder contains manifests that can be reused during cluster configuration by using the appropriate overlay transformers (more about this later).
+The [profile-base](./profile-base) is divided to the several parts based on their commonality. The content is described in detail in [profile-base README](./profile-base/README.md)
+
+### __clusters__
+The [clusters](./clusters) folder contains the actual network tree of the cluster configurations. This is the only directory that is expected to be modified by partners for deployment or maintenance.
+#### On clusters and sites
+The scope of this work is limited to Kubernetes cluster configuration, so clusters are the main building blocks of this network model. However, a notion of sites can be also important to mobile network operators.
+
+The relation between clusters and sites can be complicated. There are large sites, such as data centers, that can accommodate several clusters. At the same time there could be clusters spanning over several physical sites with remote worker nodes, or geo-diverse deployments. There is no separate site notation in the structure above - site names are expected to be encoded in the cluster names. Remote worker node configurations can be optionally placed under the "remote-sites" subdirectory for clarity. In any case, application of a particular configuration to a particular cluster is expected to be handled by tools external to Kubernetes, on the upper orchestration level.
```

Comment:
The statement `site names are expected to be encoded in the cluster names` seems to imply that a site will always equal a cluster. I think this is misleading since a cluster can span many physical sites as you describe in the RWN case. For that reason I think we should avoid correlating sites and clusters. Something along the lines of:

> A cluster may serve one or more cell towers/locations/physical sites depending on a carrier's deployment model. Cluster naming is entirely under the control of the carrier but will be used by Zero Touch Provisioning to correlate the cluster to the configuration defined within this repository.

Reply:
Because of the non-trivial relationship between a control plane of a cluster and sites reporting to it, using a `remote-sites` directory might not communicate the intended separation of concerns for the node pools. The same challenge will happen when a regional or local datacenter is used as an aggregation for multiple roles. In these setups, CU pools and DU pools can be in the same physical location using different types of pools (MCPs). In this case, the pools will be identified with different names (e.g. `ldc1-ran-cu-dell01-profile2` and `ldc1-ran-du-smci00-porfile0`) even when at the same physical location.

Reply:
@wcabanba
In my example, the pools are identified by different names (du-dc-m1, and there could be du-dc-m2 and so on for all the variants). I added the RWN example in the `remote-sites` to emphasize that it is a different site. The RWN could have been sitting in the same directory from the  MCP perspective, as it belongs to the same cluster. But since the cluster was named using CLLI, which is a site attribute, this could have been logically wrong.
Does it make sense?

Reply:
> The statement `site names are expected to be encoded in the cluster names` seems to imply that a site will always equal a cluster. I think this is misleading since a cluster can span many physical sites as you describe in the RWN case. For that reason I think we should avoid correlating sites and clusters. Something along the lines of:
> 
> > A cluster may serve one or more cell towers/locations/physical sites depending on a carrier's deployment model. Cluster naming is entirely under the control of the carrier but will be used by Zero Touch Provisioning to correlate the cluster to the configuration defined within this repository.

@imiller0 thanks, fixing this text to be more palatable. Or do you think we should totally avoid the mention of sites in this structure? I remember we agreed on just the opposite in one of ztp working sessions, but I'm ready to reconsider since it will simplify things a lot